04-ca-certificates.md

```diff
@@ -4,60 +4,54 @@ Now that the [hub-spoke network is provisioned](./03-networking.md), you can fol
-   APPGW_IP_RESOURCE_ID_03=$(az deployment group show -g rg-enterprise-networking-spokes -n  spoke-BU0001A0042-03 --query properties.outputs.appGatewayPublicIp.value -o tsv)
-   APPGW_IP_RESOURCE_ID_04=$(az deployment group show -g rg-enterprise-networking-spokes -n  spoke-BU0001A0042-04 --query properties.outputs.appGatewayPublicIp.value -o tsv)
-
-   ## Get the subdomain names selected by the script
-   CLUSTER_SUBDOMAIN_03=$(az deployment group show -g rg-enterprise-networking-spokes -n  spoke-BU0001A0042-03 --query properties.outputs.subdomainName.value -o tsv)
-   CLUSTER_SUBDOMAIN_04=$(az deployment group show -g rg-enterprise-networking-spokes -n  spoke-BU0001A0042-04 --query properties.outputs.subdomainName.value -o tsv)
-   ```
+   > :book: The Contoso Bicycle organization has an important policy that every internet-facing endpoint exposed over the Https protocol must use a trusted CA certificate, and it is not allowed to share a common wildcard certificate between them. Therefore, the organization needs to procure individual trusted CA certificates for all their Public Ips FQDN in the different regions. The Azure Application Gateway instances are going to be serving these certificates in front of every region they get deployed to.
 
    ```bash
-   ./certificate-generation/generate-cert.sh eastus2 $CLUSTER_SUBDOMAIN_03 $APPGW_FQDN_BU0001A0042_03 $APPGW_IP_RESOURCE_ID_03
-   ./certificate-generation/generate-cert.sh centralus $CLUSTER_SUBDOMAIN_04 $APPGW_FQDN_BU0001A0042_04 $APPGW_IP_RESOURCE_ID_04
+   # get the Public IP FQDNs, their subdomins, and resource ids
+   APPGW_FQDN_BU0001A0042_03=$(az deployment group show -g rg-enterprise-networking-spokes -n spoke-BU0001A0042-03 --query properties.outputs.appGwFqdn.value -o tsv)
+   APPGW_FQDN_BU0001A0042_04=$(az deployment group show -g rg-enterprise-networking-spokes -n spoke-BU0001A0042-04 --query properties.outputs.appGwFqdn.value -o tsv)
+   APPGW_SUBDOMAIN_BU0001A0042_03=$(az deployment group show -g rg-enterprise-networking-spokes -n spoke-BU0001A0042-03 --query properties.outputs.subdomainName.value -o tsv)
+   APPGW_SUBDOMAIN_BU0001A0042_04=$(az deployment group show -g rg-enterprise-networking-spokes -n spoke-BU0001A0042-04 --query properties.outputs.subdomainName.value -o tsv)
+   APPGW_IP_RESOURCE_ID_03=$(az deployment group show -g rg-enterprise-networking-spokes -n spoke-BU0001A0042-03 --query properties.outputs.appGatewayPublicIp.value -o tsv)
+   APPGW_IP_RESOURCE_ID_04=$(az deployment group show -g rg-enterprise-networking-spokes -n spoke-BU0001A0042-04 --query properties.outputs.appGatewayPublicIp.value -o tsv)
+
+   # call the Let's Encrypt certificate generation script for both PIP FQDNs
+   sudo chmod +x ./certs/letsencrypt-pip-cert-generation.sh
+   sudo chmod +x ./certs/autogenerator.sh
+   ./certs/letsencrypt-pip-cert-generation.sh $APPGW_SUBDOMAIN_BU0001A0042_03 $APPGW_FQDN_BU0001A0042_03 $APPGW_IP_RESOURCE_ID_03 eastus2
```

Comment:
Does certbot need to be installed at this point?  (Maybe that was a pre-req that I missed)

Reply:
Thanks for adding. Resolving.

Reply:
added | Addressed from [65a91](https://github.com/mspnp/aks-secure-baseline/pull/184/commits/65a9104e7ef5f6e97caf0774a8cb4424a50ce734)